Assign a name to iframes when loading the initial about:blank

notriddle opened the pull request:
Before, it would assign the name too late, causing scripts (which will not wait for another tick) to accidentally spawn pop-up windows instead of loading into the iframe.

---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data: -->
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #21886
- [x] There are tests for these changes

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/21976)
<!-- Reviewable:end -->

notriddle committed:
Assign a name to iframes when loading the initial about:blank

Before, it would assign the name too late,
causing scripts (which will not wait for another tick)
to accidentally spawn pop-up windows instead of loading
into the iframe.

highfive commented:
Heads up! This PR modifies the following files:
 * @asajeffrey: components/script/dom/htmliframeelement.rs
 * @kichjang: components/script/dom/htmliframeelement.rs

servo-wpt-sync commented:
Opened new PR for upstreamable changes.

Completed upstream sync of web-platform-test changes at https://github.com/web-platform-tests/wpt/pull/13585.

notriddle commented:
# Warning: the test case in this PR doesn't pass yet

This PR allows slither.io to be opened and played.

However, due to a second bug in the iframe implementation, the `load` event handler which performs the assertion is being called too early. It should be called only when the form is finished, but is instead being called by the initial `about:blank` document.

servo-wpt-sync commented:
Transplanted upstreamable changes to existing PR.

Completed upstream sync of web-platform-test changes at w3c/web-platform-tests#13585.

servo-wpt-sync commented:
Transplanted upstreamable changes to existing PR.

Completed upstream sync of web-platform-test changes at w3c/web-platform-tests#13585.

notriddle commented:
@bors-servo try=wpt

bors-servo commented:
:hourglass: Trying commit eafcd91760afbd0287b31aa6215c9528f6275b0d with merge 5dbf83a035dcd2d83b451b635e39be86fcb79c40...

bors-servo commented:
:broken_heart: Test failed - [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/10283)

notriddle commented:
```
{"status": "TIMEOUT", "group": "default", "message": "Test timed out", "stack": null, "subtest": "Targeting nested browsing contexts", "test": "/html/browsers/windows/targeting-cross-origin-nested-browsing-contexts.html", "line": 170625, "action": "test_result", "expected": "FAIL"}
{"status": "TIMEOUT", "group": "default", "message": null, "stack": null, "subtest": null, "test": "/html/browsers/windows/targeting-cross-origin-nested-browsing-contexts.html", "line": 170626, "action": "test_result", "expected": "OK"}
```

notriddle committed:
Skip pop-up centric WPT test

It was already broken, since Servo doesn't support pop-up windows.
The iframe changes just made it broken in a different way.

servo-wpt-sync commented:
Transplanted upstreamable changes to existing PR.

Completed upstream sync of web-platform-test changes at w3c/web-platform-tests#13585.

notriddle commented:
@bors-servo try=wpt

bors-servo commented:
:hourglass: Trying commit 7270333487ace93b260205e0fddc252299ee4718 with merge ff8cadf906b4102be2055bdb6b6d6afe26503bf5...

bors-servo commented:
:sunny: Test successful - [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/10277), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/10303)
State: approved= try=True

jdm commented on tests/wpt/web-platform-tests/html/semantics/forms/form-submission-target/form-target-iframe.html:
```diff
@@ -0,0 +1,29 @@
+<script src="/resources/testharnessreport.js"></script>
+<script>
+async_test(function(t) {
+  window.addEventListener("load", function() {
```
This should pass `t.step_func(function() {` instead.

jdm commented on tests/wpt/web-platform-tests/html/semantics/forms/form-submission-target/form-target-iframe.html:
```diff
@@ -0,0 +1,29 @@
+    form.appendChild(input);
+    document.documentElement.appendChild(form);
+    form.submit();
+    frame.addEventListener("load", function() {
```
This should also pass `t.step_func(function() {`

jdm commented on tests/wpt/web-platform-tests/html/semantics/forms/form-submission-target/form-target-iframe.html:
```diff
@@ -0,0 +1,29 @@
+    document.documentElement.appendChild(form);
+    form.submit();
+    frame.addEventListener("load", function() {
+      if (frame.contentWindow.location.href != "about:blank") {
```
Could we make this check for the actual expected value instead? Maybe using `new URL("form-target-iframe-helper.py", window.location)`?

jdm reviewed (changes requested):
Thanks for looking into this so carefully!

notriddle committed:
Use `t.step_func` and expected path checking

servo-wpt-sync commented:
Transplanted upstreamable changes to existing PR.

Completed upstream sync of web-platform-test changes at w3c/web-platform-tests#13585.

jdm commented:
@bors-servo r+

bors-servo commented:
:pushpin: Commit 10442ae has been approved by `jdm`

<!-- @bors-servo r=jdm 10442ae894004ac18847793bc11bfa1c4653cb79 -->

bors-servo commented:
:hourglass: Testing commit 10442ae894004ac18847793bc11bfa1c4653cb79 with merge 2f8dc655191455c285d4b4b5952d493fd4dc0782...

bors-servo commented:
:broken_heart: Test failed - [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/10348)

notriddle commented:
@bors-servo retry

Out-of-memory

bors-servo commented:
:zap: Previous build results for [android](http://build.servo.org/builders/android/builds/12948), [android-mac](http://build.servo.org/builders/android-mac/builds/278), [android-x86](http://build.servo.org/builders/android-x86/builds/1065), [arm32](http://build.servo.org/builders/arm32/builds/13124), [arm64](http://build.servo.org/builders/arm64/builds/13161), [linux-dev](http://build.servo.org/builders/linux-dev/builds/13245), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/10317), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/13238), [mac-rel-css2](http://build.servo.org/builders/mac-rel-css2/builds/5716), [mac-rel-wpt3](http://build.servo.org/builders/mac-rel-wpt3/builds/5795), [status-taskcluster](https://tools.taskcluster.net/task-group-inspector/#/PVHXRSSARRWL8c3FxfdO8A), [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/8617) are reusable. Rebuilding only [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/10348), [mac-rel-css1](http://build.servo.org/builders/mac-rel-css1/builds/5711), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/9982), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/10028), [mac-rel-wpt4](http://build.servo.org/builders/mac-rel-wpt4/builds/5722)...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/12948), [android-mac](http://build.servo.org/builders/android-mac/builds/278), [android-x86](http://build.servo.org/builders/android-x86/builds/1065), [arm32](http://build.servo.org/builders/arm32/builds/13124), [arm64](http://build.servo.org/builders/arm64/builds/13161), [linux-dev](http://build.servo.org/builders/linux-dev/builds/13245), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/10317), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/10349), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/13238), [mac-rel-css1](http://build.servo.org/builders/mac-rel-css1/builds/5711), [mac-rel-css2](http://build.servo.org/builders/mac-rel-css2/builds/5716), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/9982), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/10028), [mac-rel-wpt3](http://build.servo.org/builders/mac-rel-wpt3/builds/5795), [mac-rel-wpt4](http://build.servo.org/builders/mac-rel-wpt4/builds/5722), [status-taskcluster](https://tools.taskcluster.net/task-group-inspector/#/PVHXRSSARRWL8c3FxfdO8A), [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/8617)
Approved by: jdm
Pushing 2f8dc655191455c285d4b4b5952d493fd4dc0782 to master...
